{"nodes":[{"pos":[698,791],"content":"Provides data for the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridView.DataError\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> event.","needQuote":true,"needEscape":true,"source":"Provides data for the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref> event."},{"pos":[804,2004],"content":"Handling the <xref:System.Windows.Forms.DataGridView.DataError> event lets you handle exceptions thrown by code outside your control (for example, by an external data source). Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> property to determine the state of the <xref:System.Windows.Forms.DataGridView> at the time of the exception. Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property to retrieve the exception data. If you want to handle the exception by additional event handlers, set the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> property to `true`.       The <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> and <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> properties normally indicate the cell in which the data error occurred. When the error occurs in an external data source, however, the data source may not provide the column in which the error occurred. In this case, the <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> property typically indicates the column of the current cell at the time of the error.","needQuote":false,"needEscape":true,"extradata":"MT","nodes":[{"content":"Handling the &lt;xref:System.Windows.Forms.DataGridView.DataError&gt; event lets you handle exceptions thrown by code outside your control (for example, by an external data source).","pos":[0,175],"source":"Handling the <xref:System.Windows.Forms.DataGridView.DataError> event lets you handle exceptions thrown by code outside your control (for example, by an external data source)."},{"content":"Use the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A&gt; property to determine the state of the &lt;xref:System.Windows.Forms.DataGridView&gt; at the time of the exception.","pos":[176,363],"source":" Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> property to determine the state of the <xref:System.Windows.Forms.DataGridView> at the time of the exception."},{"content":"Use the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A&gt; property to retrieve the exception data.","pos":[364,484],"source":" Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property to retrieve the exception data."},{"content":"If you want to handle the exception by additional event handlers, set the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A&gt; property to <ph id=\"ph1\">`true`</ph>.","pos":[485,655],"source":" If you want to handle the exception by additional event handlers, set the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> property to `true`."},{"content":"The &lt;xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A&gt; and &lt;xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A&gt; properties normally indicate the cell in which the data error occurred.","pos":[662,888],"source":"       The <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> and <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> properties normally indicate the cell in which the data error occurred."},{"content":"When the error occurs in an external data source, however, the data source may not provide the column in which the error occurred.","pos":[889,1019]},{"content":"In this case, the &lt;xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A&gt; property typically indicates the column of the current cell at the time of the error.","pos":[1020,1198],"source":" In this case, the <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> property typically indicates the column of the current cell at the time of the error."}]},{"pos":[3980,4093],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridViewDataErrorEventArgs\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Windows.Forms.DataGridViewDataErrorEventArgs\"></xref> class."},{"pos":[4353,4381],"content":"The exception that occurred.","needQuote":true,"needEscape":true,"nodes":[{"content":"The exception that occurred.","pos":[0,28]}]},{"pos":[4448,4558],"content":"The column index of the cell that raised the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridView.DataError\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept>.","needQuote":true,"needEscape":true,"source":"The column index of the cell that raised the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref>."},{"pos":[4622,4729],"content":"The row index of the cell that raised the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridView.DataError\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept>.","needQuote":true,"needEscape":true,"source":"The row index of the cell that raised the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref>."},{"pos":[4830,4985],"content":"A bitwise combination of <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> values indicating the context in which the error occurred.","needQuote":true,"needEscape":true,"source":"A bitwise combination of <xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"></xref> values indicating the context in which the error occurred."},{"pos":[5504,5619],"content":"Gets details about the state of the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridView\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> when the error occurred.","needQuote":true,"needEscape":true,"source":"Gets details about the state of the <xref href=\"System.Windows.Forms.DataGridView\"></xref> when the error occurred."},{"pos":[5631,6406],"content":"Members of the <xref:System.Windows.Forms.DataGridViewDataErrorContexts> enumeration may be combined using the bitwise `OR` operator to represent the state of a data-bound <xref:System.Windows.Forms.DataGridView> when a data error occurred. For example, if a user enters an invalid cell value (such as entering a name in a cell that requires a date) and then selects a different cell, the <xref:System.Windows.Forms.DataGridView> will try to commit the invalid cell value. When the commit fails, the <xref:System.Windows.Forms.DataGridView> will raise a <xref:System.Windows.Forms.DataGridView.DataError> event whose Context property will have a value of <xref:System.Windows.Forms.DataGridViewDataErrorContexts> and <xref:System.Windows.Forms.DataGridViewDataErrorContexts>.","needQuote":true,"needEscape":true,"extradata":"MT","nodes":[{"content":"Members of the &lt;xref:System.Windows.Forms.DataGridViewDataErrorContexts&gt; enumeration may be combined using the bitwise <ph id=\"ph1\">`OR`</ph> operator to represent the state of a data-bound &lt;xref:System.Windows.Forms.DataGridView&gt; when a data error occurred.","pos":[0,240],"source":"Members of the <xref:System.Windows.Forms.DataGridViewDataErrorContexts> enumeration may be combined using the bitwise `OR` operator to represent the state of a data-bound <xref:System.Windows.Forms.DataGridView> when a data error occurred."},{"content":"For example, if a user enters an invalid cell value (such as entering a name in a cell that requires a date) and then selects a different cell, the &lt;xref:System.Windows.Forms.DataGridView&gt; will try to commit the invalid cell value.","pos":[241,472],"source":" For example, if a user enters an invalid cell value (such as entering a name in a cell that requires a date) and then selects a different cell, the <xref:System.Windows.Forms.DataGridView> will try to commit the invalid cell value."},{"content":"When the commit fails, the &lt;xref:System.Windows.Forms.DataGridView&gt; will raise a &lt;xref:System.Windows.Forms.DataGridView.DataError&gt; event whose Context property will have a value of &lt;xref:System.Windows.Forms.DataGridViewDataErrorContexts&gt; and &lt;xref:System.Windows.Forms.DataGridViewDataErrorContexts&gt;.","pos":[473,775],"source":" When the commit fails, the <xref:System.Windows.Forms.DataGridView> will raise a <xref:System.Windows.Forms.DataGridView.DataError> event whose Context property will have a value of <xref:System.Windows.Forms.DataGridViewDataErrorContexts> and <xref:System.Windows.Forms.DataGridViewDataErrorContexts>."}]},{"pos":[7271,7434],"content":"A bitwise combination of the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> values that specifies the context in which the error occurred.","needQuote":true,"needEscape":true,"source":"A bitwise combination of the <xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"></xref> values that specifies the context in which the error occurred."},{"pos":[7965,8010],"content":"Gets the exception that represents the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the exception that represents the error.","pos":[0,45]}]},{"pos":[8022,8287],"content":"To handle the exception by more than one &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventHandler&gt;, throw the exception again at the end of a handler by setting the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A&gt; property to <ph id=\"ph1\">`true`</ph>.","needQuote":true,"needEscape":true,"extradata":"MT","source":"To handle the exception by more than one <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>, throw the exception again at the end of a handler by setting the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> property to `true`."},{"pos":[9074,9127],"content":"An <xref:System.Exception> that represents the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"An &lt;xref:System.Exception&gt; that represents the error.","pos":[0,53],"source":"An <xref:System.Exception> that represents the error."}]},{"pos":[9685,9863],"content":"Gets or sets a value indicating whether to throw the exception after the <bpt id=\"p1\">&lt;xref href=\"System.Windows.Forms.DataGridViewDataErrorEventHandler\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> delegate is finished with it.","needQuote":true,"needEscape":true,"source":"Gets or sets a value indicating whether to throw the exception after the <xref href=\"System.Windows.Forms.DataGridViewDataErrorEventHandler\"></xref> delegate is finished with it."},{"pos":[9876,10457],"content":"Set this property to `false` if the <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> has dealt with the exception represented by the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property, and you do not want to propagate the exception to another <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> that may handle the <xref:System.Windows.Forms.DataGridView.DataError> event.       For debugging purposes, it can be useful to propagate the exception and inspect the stack trace for information regarding the context of the error.","needQuote":false,"needEscape":true,"extradata":"MT","nodes":[{"content":"Set this property to <ph id=\"ph1\">`false`</ph> if the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventHandler&gt; has dealt with the exception represented by the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A&gt; property, and you do not want to propagate the exception to another &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventHandler&gt; that may handle the &lt;xref:System.Windows.Forms.DataGridView.DataError&gt; event.","pos":[0,425],"source":"Set this property to `false` if the <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> has dealt with the exception represented by the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property, and you do not want to propagate the exception to another <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> that may handle the <xref:System.Windows.Forms.DataGridView.DataError> event."},{"content":"For debugging purposes, it can be useful to propagate the exception and inspect the stack trace for information regarding the context of the error.","pos":[432,579]}]},{"pos":[11302,11551],"content":"<xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref> if the exception should be thrown; otherwise, <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>. The default is <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>.","needQuote":true,"needEscape":true,"nodes":[{"content":"<bpt id=\"p1\">&lt;xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> if the exception should be thrown; otherwise, <bpt id=\"p2\">&lt;xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"&gt;</bpt><ept id=\"p2\">&lt;/xref&gt;</ept>.","pos":[0,170],"source":"<xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref> if the exception should be thrown; otherwise, <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>."},{"content":"The default is <bpt id=\"p1\">&lt;xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept>.","pos":[171,249],"source":" The default is <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>."}]},{"pos":[11740,11985],"content":"When setting this property to <bpt id=\"p1\">&lt;xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept>, the &lt;xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*&gt; property value is <bpt id=\"p2\">&lt;xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"&gt;</bpt><ept id=\"p2\">&lt;/xref&gt;</ept>.","needQuote":true,"needEscape":true,"source":"When setting this property to <xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref>, the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*> property value is <xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"></xref>."}],"content":"### YamlMime:ManagedReference\nitems:\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  id: DataGridViewDataErrorEventArgs\n  children:\n  - System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)\n  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Context\n  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception\n  - System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException\n  langs:\n  - csharp\n  name: DataGridViewDataErrorEventArgs\n  nameWithType: DataGridViewDataErrorEventArgs\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  type: Class\n  summary: Provides data for the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref> event.\n  remarks: \"Handling the <xref:System.Windows.Forms.DataGridView.DataError> event lets you handle exceptions thrown by code outside your control (for example, by an external data source). Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> property to determine the state of the <xref:System.Windows.Forms.DataGridView> at the time of the exception. Use the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property to retrieve the exception data. If you want to handle the exception by additional event handlers, set the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> property to `true`.  \\n  \\n The <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> and <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> properties normally indicate the cell in which the data error occurred. When the error occurs in an external data source, however, the data source may not provide the column in which the error occurred. In this case, the <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> property typically indicates the column of the current cell at the time of the error.\"\n  example:\n  - \"The following code example demonstrates how to respond to information provided by the DataGridViewDataErrorEventArgs class. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \\n  \\n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/t-system.windows.forms.d_56_1.cpp)]\\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/t-system.windows.forms.d_56_1.cs)]\\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/t-system.windows.forms.d_56_1.vb)]\"\n  syntax:\n    content: 'public class DataGridViewDataErrorEventArgs : System.Windows.Forms.DataGridViewCellCancelEventArgs'\n  inheritance:\n  - System.ComponentModel.CancelEventArgs\n  - System.Windows.Forms.DataGridViewCellCancelEventArgs\n  implements: []\n  inheritedMembers:\n  - System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex\n  - System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex\n  platform:\n  - net462\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)\n  id: '#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)'\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  langs:\n  - csharp\n  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n  type: Constructor\n  assemblies:\n  - System.Windows.Forms\n  namespace: System.Windows.Forms\n  summary: Initializes a new instance of the <xref href=\"System.Windows.Forms.DataGridViewDataErrorEventArgs\"></xref> class.\n  syntax:\n    content: public DataGridViewDataErrorEventArgs (Exception exception, int columnIndex, int rowIndex, System.Windows.Forms.DataGridViewDataErrorContexts context);\n    parameters:\n    - id: exception\n      type: System.Exception\n      description: The exception that occurred.\n    - id: columnIndex\n      type: System.Int32\n      description: The column index of the cell that raised the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref>.\n    - id: rowIndex\n      type: System.Int32\n      description: The row index of the cell that raised the <xref href=\"System.Windows.Forms.DataGridView.DataError\"></xref>.\n    - id: context\n      type: System.Windows.Forms.DataGridViewDataErrorContexts\n      description: A bitwise combination of <xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"></xref> values indicating the context in which the error occurred.\n  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context\n  id: Context\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  langs:\n  - csharp\n  name: Context\n  nameWithType: DataGridViewDataErrorEventArgs.Context\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context\n  type: Property\n  assemblies:\n  - System.Windows.Forms\n  namespace: System.Windows.Forms\n  summary: Gets details about the state of the <xref href=\"System.Windows.Forms.DataGridView\"></xref> when the error occurred.\n  remarks: Members of the <xref:System.Windows.Forms.DataGridViewDataErrorContexts> enumeration may be combined using the bitwise `OR` operator to represent the state of a data-bound <xref:System.Windows.Forms.DataGridView> when a data error occurred. For example, if a user enters an invalid cell value (such as entering a name in a cell that requires a date) and then selects a different cell, the <xref:System.Windows.Forms.DataGridView> will try to commit the invalid cell value. When the commit fails, the <xref:System.Windows.Forms.DataGridView> will raise a <xref:System.Windows.Forms.DataGridView.DataError> event whose Context property will have a value of <xref:System.Windows.Forms.DataGridViewDataErrorContexts> and <xref:System.Windows.Forms.DataGridViewDataErrorContexts>.\n  example:\n  - \"The following code example demonstrates how to investigate the error context. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \\n  \\n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_134_1.cpp)]\\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_134_1.cs)]\\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_134_1.vb)]\"\n  syntax:\n    content: public System.Windows.Forms.DataGridViewDataErrorContexts Context { get; }\n    return:\n      type: System.Windows.Forms.DataGridViewDataErrorContexts\n      description: A bitwise combination of the <xref href=\"System.Windows.Forms.DataGridViewDataErrorContexts\"></xref> values that specifies the context in which the error occurred.\n  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception\n  id: Exception\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  langs:\n  - csharp\n  name: Exception\n  nameWithType: DataGridViewDataErrorEventArgs.Exception\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception\n  type: Property\n  assemblies:\n  - System.Windows.Forms\n  namespace: System.Windows.Forms\n  summary: Gets the exception that represents the error.\n  remarks: To handle the exception by more than one <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>, throw the exception again at the end of a handler by setting the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> property to `true`.\n  example:\n  - \"The following code example demonstrates checking the Exception property. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \\n  \\n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_249_1.cpp)]\\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_249_1.cs)]\\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_249_1.vb)]\"\n  syntax:\n    content: public Exception Exception { get; }\n    return:\n      type: System.Exception\n      description: An <xref:System.Exception> that represents the error.\n  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException\n  id: ThrowException\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  langs:\n  - csharp\n  name: ThrowException\n  nameWithType: DataGridViewDataErrorEventArgs.ThrowException\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException\n  type: Property\n  assemblies:\n  - System.Windows.Forms\n  namespace: System.Windows.Forms\n  summary: Gets or sets a value indicating whether to throw the exception after the <xref href=\"System.Windows.Forms.DataGridViewDataErrorEventHandler\"></xref> delegate is finished with it.\n  remarks: \"Set this property to `false` if the <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> has dealt with the exception represented by the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> property, and you do not want to propagate the exception to another <xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> that may handle the <xref:System.Windows.Forms.DataGridView.DataError> event.  \\n  \\n For debugging purposes, it can be useful to propagate the exception and inspect the stack trace for information regarding the context of the error.\"\n  example:\n  - \"The following code example demonstrates using the ThrowException property to indicate that the exception should not be thrown. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \\n  \\n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_202_1.cpp)]\\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_202_1.cs)]\\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_202_1.vb)]\"\n  syntax:\n    content: public bool ThrowException { get; set; }\n    return:\n      type: System.Boolean\n      description: <xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref> if the exception should be thrown; otherwise, <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>. The default is <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>.\n  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*\n  exceptions:\n  - type: System.ArgumentException\n    commentId: T:System.ArgumentException\n    description: When setting this property to <xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref>, the <xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*> property value is <xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"></xref>.\n  platform:\n  - net462\nreferences:\n- uid: System.Windows.Forms.DataGridViewCellCancelEventArgs\n  isExternal: false\n  name: System.Windows.Forms.DataGridViewCellCancelEventArgs\n- uid: System.ArgumentException\n  isExternal: true\n  name: System.ArgumentException\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)\n- uid: System.Exception\n  parent: System\n  isExternal: true\n  name: Exception\n  nameWithType: Exception\n  fullName: System.Exception\n- uid: System.Int32\n  parent: System\n  isExternal: true\n  name: Int32\n  nameWithType: Int32\n  fullName: System.Int32\n- uid: System.Windows.Forms.DataGridViewDataErrorContexts\n  parent: System.Windows.Forms\n  isExternal: false\n  name: DataGridViewDataErrorContexts\n  nameWithType: DataGridViewDataErrorContexts\n  fullName: System.Windows.Forms.DataGridViewDataErrorContexts\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: Context\n  nameWithType: DataGridViewDataErrorEventArgs.Context\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: Exception\n  nameWithType: DataGridViewDataErrorEventArgs.Exception\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: ThrowException\n  nameWithType: DataGridViewDataErrorEventArgs.ThrowException\n  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException\n- uid: System.Boolean\n  parent: System\n  isExternal: true\n  name: Boolean\n  nameWithType: Boolean\n  fullName: System.Boolean\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: DataGridViewDataErrorEventArgs\n  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: Context\n  nameWithType: DataGridViewDataErrorEventArgs.Context\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: Exception\n  nameWithType: DataGridViewDataErrorEventArgs.Exception\n- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*\n  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs\n  isExternal: false\n  name: ThrowException\n  nameWithType: DataGridViewDataErrorEventArgs.ThrowException\n"}